classes/PodsRESTHandlers.php

```diff
@@ -0,0 +1,195 @@
+	 * @param $id
+	 *
+	 * @return bool|\Pods
+	 */
+	protected static function get_pod( $pod_name, $id ) {
+		if ( ! self::$pod || self::$pod->pod != $pod_name ) {
+			self::$pod = pods( $pod_name, $id, true );
+		}
+
+		if ( self::$pod && self::$pod->id != $id ) {
+			self::$pod->fetch( $id );
+		}
+
+		return self::$pod;
+
+	}
+
+	/**
+	 * Handler for getting custom field data.
+	 *
+	 * @since 2.5.6
+	 *
+	 * @param array $object The object from the response
+	 * @param string $field_name Name of field
+	 * @param \WP_REST_Request $request Current request
+	 *
+	 * @return mixed
+	 */
+	public static function get_handler( $object, $field_name, $request ) {
+		$pod_name = pods_v( 'type', $object );
+		$id = pods_v( 'id', $object );
+		$pod = self::get_pod( $pod_name, $id );
+		if ( $pod && pods_rest_api_field_allowed_to_extend( $field_name, $pod ) ) {
+			$params = null;
+			$field_data = $pod->fields( $field_name );
+			if( 'pick' == pods_v( 'type', $field_data ) ) {
+				$output_type = pods_v( 'rest_pick_response', $field_data['options'], 'array' );
+				if(  'array' == $output_type ) {
+
+					$related_pod = $pod->field( $field_name, array( 'output' => 'pod' ) );
+					$fields = $related_pod->fields();
+					$fields = array_keys( $fields );
+					if ( isset( $related_pod->pod_data[ 'object_fields' ] ) && ! empty( $related_pod->pod_data[ 'object_fields' ] ) ) {
+						$fields = array_merge( $fields, array_keys( $related_pod->pod_data[ 'object_fields' ] ) );
+					}
+
+					/**
+					 * What fields to show in a related field REST response.
+					 *
+					 * @since 0.0.1
+					 *
+					 * @param array $fields The fields to show
+					 * @param string $field_name The name of the field
+					 * @param object|\Pods $pod The Pods object for Pod relationship is from.
+					 * @param object|\Pods $pod The Pods object for Pod relationship is to.
+					 * @param int $id Current item ID
+					 * @param object||WP_REST_Request Current request object.
+					 */
+					$fields = apply_filters( 'pods_rest_api_fields_for_relationship_response', $fields, $field_name, $pod, $related_pod, $id, $request );
+
+					$depth = pods_v( 'rest_pick_depth', $field_data[ 'options' ], 2 );
+
+					/**
+					 * What depth to use for a related field REST response.
+					 *
+					 * @since 0.0.1
+					 *
+					 * @param array $depth The depth.
+					 * @param string $field_name The name of the field
+					 * @param object|\Pods $pod The Pods object for Pod relationship is from.
+					 * @param object|\Pods $pod The Pods object for Pod relationship is to.
+					 * @param int $id Current item ID
+					 * @param object||WP_REST_Request Current request object.
+					 */
+					$depth = apply_filters( 'pods_rest_api_depth_for_relationship_response', $depth, $field_name, $pod, $related_pod, $id, $request );
+
+					$params = array(
+						'fields' => $fields,
+						'depth' => $depth
+
+					);
+
+					$data = $pod->api->export_pod_item( $related_pod, $params );
```

Comment:
with the current implementation this can lead to huge loading times due to the way it's currently implemented iterating over and over the same pods objects especially if someone increases the depth ...